sql/src/main/java/io/crate/planner/statement/CopyStatementPlanner.java

```diff
@@ -68,124 +67,150 @@ public CopyStatementPlanner(ClusterService clusterService) {
         this.clusterService = clusterService;
     }
 
-    public Plan planCopyFrom(CopyFromAnalyzedStatement analysis, Planner.Context context) {
+    public Plan planCopyFrom(CopyFromAnalyzedStatement copyFrom, Planner.Context context) {
         /*
-         * copy from has two "modes":
-         *
-         * 1: non-partitioned tables or partitioned tables with partition ident --> import into single es index
-         *    -> collect raw source and import as is
-         *
-         * 2: partitioned table without partition ident
-         *    -> collect document and partition by values
-         *    -> exclude partitioned by columns from document
-         *    -> insert into es index (partition determined by partition by value)
+         * Create a plan that reads json-objects-lines from a file and then executes upsert requests to index the data
          */
-
-        DocTableInfo table = analysis.table();
-        int clusteredByPrimaryKeyIdx = table.primaryKey().indexOf(analysis.table().clusteredBy());
-        List<String> partitionedByNames;
-        String partitionIdent = null;
-
-        List<BytesRef> partitionValues;
-        if (analysis.partitionIdent() == null) {
-
+        DocTableInfo table = copyFrom.table();
+        String partitionIdent = copyFrom.partitionIdent();
+        List<String> partitionedByNames = Collections.emptyList();
+        List<BytesRef> partitionValues = Collections.emptyList();
+        if (partitionIdent == null) {
             if (table.isPartitioned()) {
-                partitionedByNames = Lists.newArrayList(
-                    Lists.transform(table.partitionedBy(), ColumnIdent::fqn));
-            } else {
-                partitionedByNames = Collections.emptyList();
+                partitionedByNames = Lists2.copyAndReplace(table.partitionedBy(), ColumnIdent::fqn);
             }
-            partitionValues = ImmutableList.of();
         } else {
             assert table.isPartitioned() : "table must be partitioned if partitionIdent is set";
             // partitionIdent is present -> possible to index raw source into concrete es index
-
-            partitionValues = PartitionName.decodeIdent(analysis.partitionIdent());
-            partitionIdent = analysis.partitionIdent();
-            partitionedByNames = Collections.emptyList();
+            partitionValues = PartitionName.decodeIdent(partitionIdent);
         }
 
+        // no need to collect _id columns -> it's auto generated anyway
```

Comment:
Comment isn't accurate, _id columns are not exported so they likely can't be collected